### DISCUSSION TITLE
How to change TR HOVER COLOR in TABLE?

### DISCUSSION THREAD

**Discussion by phantodev** [heroui-inc/heroui#3153] [Help]
Hello guys... i tried everything to change my TR HOVER COLOR in in TABLE COMPONENT... Could you help me?

classNames={{
            th: "dark:bg-dark dark:text-white",
            tr: "hover:bg-red-500" // for example
}}

But i can see a red color and another gray overlay over my red color :-(

If you know of a way for me to change this globally in all my tables I would love it. Maybe some theme configuration for the table in general.

Thank you so much guys

**Comment by alphaxek** [comment]
Hi @phantodev, you want to change the text color or background color?

**Reply by kipavy** [reply]
Hi, I have exactly the same issue, I want to change hover color on rows but I can't override the default one...

**Comment by MarcusDelvecchio** [comment]
same issue here, any solutions?

**Comment by AnYiEE** [comment]
Because this style is on each `td` by default, it is `group-aria-[selected=false]:group-data-[hover=true]:before:bg-default-100 group-aria-[selected=false]:group-data-[hover=true]:before:opacity-70`.

**Comment by erickmoore** [comment]
@phantodev you are likely seeing those corners because the TableCell (td) also needs to be rounded when you are overwriting the style:

```javascript
<TableRow
     key={item.key}
      className="hover:bg-blue-500 dark:hover:bg-blue-500 first:rounded-l-lg last:rounded-r-lg">
              {(columnKey: string | number) => (
                <TableCell className="first:rounded-l-lg last:rounded-r-lg">
                  {renderCell(item, columnKey)}
                </TableCell>
              )}
</TableRow>
```

**Comment by thatrafi** [comment]
Any solution? it's 2025 i still can't change the hover color

`tr: clsx(
        'h-[33px]',
        // default hero ui hover is 0.9 opacity but we don't want that here
        'hover:!bg-[var(--color-surface-container-surface-container-high)]',
        'data-[selected=true]:!bg-[var(--color-surface-container-surface-container-high)]',
        'border-b border-[var(--color-surface-outline)]',
        'first:rounded-t-[var(--radius-sm)] last:rounded-b-[var(--radius-sm)]',
      ),
      td: clsx(
        // first
        'first:group-data-[first=true]/tr:before:rounded-none',
        'last:group-data-[first=true]/tr:before:rounded-none',
        // middle
        'group-data-[middle=true]/tr:before:rounded-none',
        // last
        'first:group-data-[last=true]/tr:before:rounded-none',
        'last:group-data-[last=true]/tr:before:rounded-none',
      ),
`

**Comment by Razvan200526** [comment]
the solution i found is basically to write normal css, i used it for Select component , where i had <Select classNames={{base : 'hover:bg-something'}} (...other properties)/> and i had to use the style prop like <Select style={styles.heroBg} .../> 
```
//filename.module.css
.heroBg {
  background: var(--color-background);
  transition: background 0.2s;
}
.heroBg:hover {
  background: color-mix(in oklch, var(--color-secondary) 10%, transparent);
}
```
i defined this file and imported it into my compoenent files and it worked.Only way i could find it to work
